vmcnet/train/runners.py

```diff
@@ -43,20 +44,39 @@
 def _get_logdir_and_save_config(reload_config: ConfigDict, config: ConfigDict) -> str:
     logging.info("Reload configuration: \n%s", reload_config)
     logging.info("Running with configuration: \n%s", config)
-    if config.logdir:
-        if config.save_to_current_datetime_subfolder:
-            dirname = utils.io.add_suffix_for_uniqueness(
-                name=datetime.datetime.now().strftime("%Y%m%d-%H%M%S"),
-                logdir=config.logdir,
-            )
-            config.logdir = os.path.join(config.logdir, dirname)
 
-        logdir = config.logdir
-        utils.io.save_config_dict_to_json(config, logdir, "config")
-        utils.io.save_config_dict_to_json(reload_config, logdir, "reload_config")
-    else:
-        logdir = None
-    return logdir
+    if config.save_to_named_subfolder:
+        config.logdir = os.path.join(config.logdir, _get_name_from_config(config))
+
+    if config.save_to_current_datetime_subfolder:
+        config.logdir = os.path.join(
+            config.logdir, datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
+        )
+
+    config.logdir = utils.io.add_suffix_for_uniqueness(config.logdir)
+    utils.io.save_config_dict_to_json(config, config.logdir, "config")
+    utils.io.save_config_dict_to_json(reload_config, config.logdir, "reload_config")
+
+    logging.info("Saving to %s", config.logdir)
+    return config.logdir
+
+
+def _get_name_from_config(config: ConfigDict) -> str:
+    name = config.notes
+
+    # remove remarks in parentheses (assumes not nested)
```

Comment:
I don't love this logic here, since it seems to make it a bit opaque to the user where exactly the results will get logged. I think it might be better to just have a separate `notes` field and `dirname` field and then it's assumed that `dirname` is already a valid directory name whereas 'notes' can be anything, i.e. arbitrarily long poetic notes that you wouldn't want to make into a file name anyway. Thoughts?